### ISSUE TITLE
Keyboard detected as Xbox 360 controller

### ISSUE THREAD

**Issue by J2Kbr** [J2Kbr/TitanTwo#175]
```
[USB DESCRIPTOR]
12           bLength: 18
01           bDescriptorType: 1 - DEVICE
00 02        bcdUSB: 2.00
00           bDeviceClass: 0 (00)
00           bDeviceSubClass: 0 (00)
00           bDeviceProtocol: 0 (00)
40           bMaxPacketSize0: 64
EB 03        idVendor: 03EB
01 FF        idProduct: FF01
13 01        bcdDevice: 1.13
01           iManufacturer: 1
02           iProduct: 2
00           iSerialNumber: 0
01           bNumConfigurations: 1

09           bLength: 9
02           bDescriptorType: 2 - CONFIGURATION
CE 00        wTotalLength: 206
07           bNumInterfaces: 7
01           bConfigurationValue: 1
00           iConfiguration: 0
A0           bmAttributes: 160 (Bus Powered, Remote Wakeup)
C8           MaxPower: 400mA

09             bLength: 9
04             bDescriptorType: 4 - INTERFACE
00             bInterfaceNumber: 0
00             bAlternateSetting: 0
02             bNumEndpoints: 2
FF             bInterfaceClass: 255 (FF)
5D             bInterfaceSubClass: 93 (5D)
01             bInterfaceProtocol: 1 (01)
00             iInterface: 0

11               bLength: 17
21               bDescriptorType: 33 - HID
00 01            bcdHID: 1.00
01               bCountryCode: 1
25               bNumDescriptors: 37
81               bDescriptorType: 129
14 00            wDescriptorLength: 20
00               bDescriptorType: 0
00 00            wDescriptorLength: 0
13               bDescriptorType: 19
02 08            wDescriptorLength: 2050
00               bDescriptorType: 0
00 07            wDescriptorLength: 1792
05               bDescriptorType: 5
81 03            wDescriptorLength: 897
20               bDescriptorType: 32
00 04            wDescriptorLength: 1024
07               bDescriptorType: 7
05 02            wDescriptorLength: 517
03               bDescriptorType: 3
20 00            wDescriptorLength: 32
08               bDescriptorType: 8
09 04            wDescriptorLength: 1033
01               bDescriptorType: 1
00 01            wDescriptorLength: 256
03               bDescriptorType: 3
01 01            wDescriptorLength: 257
00               bDescriptorType: 0
09 21            wDescriptorLength: 8457
11               bDescriptorType: 17
01 00            wDescriptorLength: 1
01               bDescriptorType: 1
22 3B            wDescriptorLength: 15138
00               bDescriptorType: 0
07 05            wDescriptorLength: 1287
83               bDescriptorType: 131
03 08            wDescriptorLength: 2051
00               bDescriptorType: 0
0A 09            wDescriptorLength: 2314
04               bDescriptorType: 4
02 00            wDescriptorLength: 2
02               bDescriptorType: 2
03 00            wDescriptorLength: 3
00               bDescriptorType: 0
00 09            wDescriptorLength: 2304
21               bDescriptorType: 33
11 01            wDescriptorLength: 273
00               bDescriptorType: 0
01 22            wDescriptorLength: 8705
35               bDescriptorType: 53
00 07            wDescriptorLength: 1792
05               bDescriptorType: 5
85 03            wDescriptorLength: 901
40               bDescriptorType: 64
00 04            wDescriptorLength: 1024
07               bDescriptorType: 7
05 04            wDescriptorLength: 1029
03               bDescriptorType: 3
40 00            wDescriptorLength: 64
04               bDescriptorType: 4
09 04            wDescriptorLength: 1033
03               bDescriptorType: 3
00 01            wDescriptorLength: 256
03               bDescriptorType: 3
00 01            wDescriptorLength: 256
00               bDescriptorType: 0
09 21            wDescriptorLength: 8457
11               bDescriptorType: 17
01 00            wDescriptorLength: 1
01               bDescriptorType: 1
22 7B            wDescriptorLength: 31522
00               bDescriptorType: 0
07 05            wDescriptorLength: 1287
86               bDescriptorType: 134
03 40            wDescriptorLength: 16387
00               bDescriptorType: 0
01 09            wDescriptorLength: 2305
04               bDescriptorType: 4
04 00            wDescriptorLength: 4

07               bLength: 7
05               bDescriptorType: 5 - ENDPOINT
81               bEndpointAddress: 1 IN
03               bmAttributes: 3 (Interrupt, None, Data)
20 00            wMaxPacketSize: 32
04               bInterval: 4

07               bLength: 7
05               bDescriptorType: 5 - ENDPOINT
02               bEndpointAddress: 2 OUT
03               bmAttributes: 3 (Interrupt, None, Data)
20 00            wMaxPacketSize: 32
08               bInterval: 8

09             bLength: 9
04             bDescriptorType: 4 - INTERFACE
01             bInterfaceNumber: 1
00             bAlternateSetting: 0
01             bNumEndpoints: 1
03             bInterfaceClass: 3 (03)
01             bInterfaceSubClass: 1 (01)
01             bInterfaceProtocol: 1 (01)
00             iInterface: 0

09               bLength: 9
21               bDescriptorType: 33 - HID
11 01            bcdHID: 1.11
00               bCountryCode: 0
01               bNumDescriptors: 1
22               bDescriptorType: 34
3B 00            wDescriptorLength: 59

07               bLength: 7
05               bDescriptorType: 5 - ENDPOINT
83               bEndpointAddress: 3 IN
03               bmAttributes: 3 (Interrupt, None, Data)
08 00            wMaxPacketSize: 8
0A               bInterval: 10

09             bLength: 9
04             bDescriptorType: 4 - INTERFACE
02             bInterfaceNumber: 2
00             bAlternateSetting: 0
02             bNumEndpoints: 2
03             bInterfaceClass: 3 (03)
00             bInterfaceSubClass: 0 (00)
00             bInterfaceProtocol: 0 (00)
00             iInterface: 0

09               bLength: 9
21               bDescriptorType: 33 - HID
11 01            bcdHID: 1.11
00               bCountryCode: 0
01               bNumDescriptors: 1
22               bDescriptorType: 34
35 00            wDescriptorLength: 53

07               bLength: 7
05               bDescriptorType: 5 - ENDPOINT
85               bEndpointAddress: 5 IN
03               bmAttributes: 3 (Interrupt, None, Data)
40 00            wMaxPacketSize: 64
04               bInterval: 4

07               bLength: 7
05               bDescriptorType: 5 - ENDPOINT
04               bEndpointAddress: 4 OUT
03               bmAttributes: 3 (Interrupt, None, Data)
40 00            wMaxPacketSize: 64
04               bInterval: 4

09             bLength: 9
04             bDescriptorType: 4 - INTERFACE
03             bInterfaceNumber: 3
00             bAlternateSetting: 0
01             bNumEndpoints: 1
03             bInterfaceClass: 3 (03)
00             bInterfaceSubClass: 0 (00)
01             bInterfaceProtocol: 1 (01)
00             iInterface: 0

09               bLength: 9
21               bDescriptorType: 33 - HID
11 01            bcdHID: 1.11
00               bCountryCode: 0
01               bNumDescriptors: 1
22               bDescriptorType: 34
7B 00            wDescriptorLength: 123

07               bLength: 7
05               bDescriptorType: 5 - ENDPOINT
86               bEndpointAddress: 6 IN
03               bmAttributes: 3 (Interrupt, None, Data)
40 00            wMaxPacketSize: 64
01               bInterval: 1

09             bLength: 9
04             bDescriptorType: 4 - INTERFACE
04             bInterfaceNumber: 4
00             bAlternateSetting: 0
01             bNumEndpoints: 1
03             bInterfaceClass: 3 (03)
00             bInterfaceSubClass: 0 (00)
01             bInterfaceProtocol: 1 (01)
00             iInterface: 0

09               bLength: 9
21               bDescriptorType: 33 - HID
11 01            bcdHID: 1.11
00               bCountryCode: 0
01               bNumDescriptors: 1
22               bDescriptorType: 34
19 00            wDescriptorLength: 25

07               bLength: 7
05               bDescriptorType: 5 - ENDPOINT
87               bEndpointAddress: 7 IN
03               bmAttributes: 3 (Interrupt, None, Data)
08 00            wMaxPacketSize: 8
10               bInterval: 16

09             bLength: 9
04             bDescriptorType: 4 - INTERFACE
05             bInterfaceNumber: 5
00             bAlternateSetting: 0
01             bNumEndpoints: 1
03             bInterfaceClass: 3 (03)
00             bInterfaceSubClass: 0 (00)
00             bInterfaceProtocol: 0 (00)
00             iInterface: 0

09               bLength: 9
21               bDescriptorType: 33 - HID
11 01            bcdHID: 1.11
00               bCountryCode: 0
01               bNumDescriptors: 1
22               bDescriptorType: 34
51 00            wDescriptorLength: 81

07               bLength: 7
05               bDescriptorType: 5 - ENDPOINT
88               bEndpointAddress: 8 IN
03               bmAttributes: 3 (Interrupt, None, Data)
20 00            wMaxPacketSize: 32
08               bInterval: 8

09             bLength: 9
04             bDescriptorType: 4 - INTERFACE
06             bInterfaceNumber: 6
00             bAlternateSetting: 0
01             bNumEndpoints: 1
03             bInterfaceClass: 3 (03)
00             bInterfaceSubClass: 0 (00)
00             bInterfaceProtocol: 0 (00)
00             iInterface: 0

09               bLength: 9
21               bDescriptorType: 33 - HID
11 01            bcdHID: 1.11
00               bCountryCode: 0
01               bNumDescriptors: 1
22               bDescriptorType: 34
17 00            wDescriptorLength: 23

07               bLength: 7
05               bDescriptorType: 5 - ENDPOINT
89               bEndpointAddress: 9 IN
03               bmAttributes: 3 (Interrupt, None, Data)
40 00            wMaxPacketSize: 64
04               bInterval: 4

[STRING 1 DESCRIPTOR]
Wooting

[STRING 2 DESCRIPTOR]
WootingOne

[INTERFACE 1 HID DESCRIPTOR]
05 01        (GLOBAL) USAGE_PAGE         0x0001 Generic Desktop Page
09 06        (LOCAL)  USAGE              0x00010006 Keyboard (CA=Application Collection)
A1 01        (MAIN)   COLLECTION         0x01 Application (Usage=0x00010006: Page=Generic Desktop Page, Usage=Keyboard, Type=CA)
05 07          (GLOBAL) USAGE_PAGE         0x0007 Keyboard/Keypad Page
19 E0          (LOCAL)  USAGE_MINIMUM      0x000700E0 Keyboard Left Control (DV=Dynamic Value)
29 E7          (LOCAL)  USAGE_MAXIMUM      0x000700E7 Keyboard Right GUI (DV=Dynamic Value)
15 00          (GLOBAL) LOGICAL_MINIMUM    0x00 (0)
25 01          (GLOBAL) LOGICAL_MAXIMUM    0x01 (1)
75 01          (GLOBAL) REPORT_SIZE        0x01 (1) Number of bits per field
95 00          (GLOBAL) REPORT_COUNT       0x00 (0) Number of fields
81 02          (MAIN)   INPUT              0x00000002 (0 fields x 1 bit) 0=Data 1=Variable 0=Absolute 0=NoWrap 0=Linear 0=PrefState 0=NoNull 0=NonVolatile 0=Bitmap
81 01          (MAIN)   INPUT              0x00000001 (0 fields x 1 bit) 1=Constant 0=Array 0=Absolute 0=Ignored 0=Ignored 0=PrefState 0=NoNull
19 00          (LOCAL)  USAGE_MINIMUM      0x00070000 Keyboard No event indicated (Sel=Selector)
29 65          (LOCAL)  USAGE_MAXIMUM      0x00070065 Keyboard Application (Sel=Selector)
15 00          (GLOBAL) LOGICAL_MINIMUM    0x00 (0)
25 65          (GLOBAL) LOGICAL_MAXIMUM    0x65 (101)
75 08          (GLOBAL) REPORT_SIZE        0x08 (8) Number of bits per field
95 00          (GLOBAL) REPORT_COUNT       0x00 (0) Number of fields
81 00          (MAIN)   INPUT              0x00000000 (0 fields x 8 bits) 0=Data 0=Array 0=Absolute 0=Ignored 0=Ignored 0=PrefState 0=NoNull
05 08          (GLOBAL) USAGE_PAGE         0x0008 LED Indicator Page
19 01          (LOCAL)  USAGE_MINIMUM      0x00080001 Num Lock (OOC=On/Off Control)
29 05          (LOCAL)  USAGE_MAXIMUM      0x00080005 Kana (OOC=On/Off Control)
15 00          (GLOBAL) LOGICAL_MINIMUM    0x00 (0)
25 01          (GLOBAL) LOGICAL_MAXIMUM    0x01 (1)
75 01          (GLOBAL) REPORT_SIZE        0x01 (1) Number of bits per field
95 05          (GLOBAL) REPORT_COUNT       0x05 (5) Number of fields
91 02          (MAIN)   OUTPUT             0x00000002 (5 fields x 1 bit) 0=Data 1=Variable 0=Absolute 0=NoWrap 0=Linear 0=PrefState 0=NoNull 0=NonVolatile 0=Bitmap
95 03          (GLOBAL) REPORT_COUNT       0x03 (3) Number of fields
91 01          (MAIN)   OUTPUT             0x00000001 (3 fields x 1 bit) 1=Constant 0=Array 0=Absolute 0=NoWrap 0=Linear 0=PrefState 0=NoNull 0=NonVolatile 0=Bitmap
C0           (MAIN)   END_COLLECTION     Application

[INTERFACE 2 HID DESCRIPTOR]
06 3713      (GLOBAL) USAGE_PAGE         0x1337 Reserved
09 01        (LOCAL)  USAGE              0x13370001 
A1 01        (MAIN)   COLLECTION         0x01 Application (Usage=0x13370001: Page=Reserved, Usage=, Type=)
09 02          (LOCAL)  USAGE              0x13370002 
09 03          (LOCAL)  USAGE              0x13370003 
15 00          (GLOBAL) LOGICAL_MINIMUM    0x00 (0)
26 FF00        (GLOBAL) LOGICAL_MAXIMUM    0x00FF (255)
75 08          (GLOBAL) REPORT_SIZE        0x08 (8) Number of bits per field
95 80          (GLOBAL) REPORT_COUNT       0x80 (-128) Number of fields
81 02          (MAIN)   INPUT              0x00000002 (-128 fields x 8 bits) 0=Data 1=Variable 0=Absolute 0=NoWrap 0=Linear 0=PrefState 0=NoNull 0=NonVolatile 0=Bitmap
09 04          (LOCAL)  USAGE              0x13370004 
09 05          (LOCAL)  USAGE              0x13370005 
15 00          (GLOBAL) LOGICAL_MINIMUM    0x00 (0)
26 FF00        (GLOBAL) LOGICAL_MAXIMUM    0x00FF (255)
75 08          (GLOBAL) REPORT_SIZE        0x08 (8) Number of bits per field
95 80          (GLOBAL) REPORT_COUNT       0x80 (-128) Number of fields
91 02          (MAIN)   OUTPUT             0x00000002 (-128 fields x 8 bits) 0=Data 1=Variable 0=Absolute 0=NoWrap 0=Linear 0=PrefState 0=NoNull 0=NonVolatile 0=Bitmap
09 06          (LOCAL)  USAGE              0x13370006 
09 07          (LOCAL)  USAGE              0x13370007 
15 00          (GLOBAL) LOGICAL_MINIMUM    0x00 (0)
26 FF00        (GLOBAL) LOGICAL_MAXIMUM    0x00FF (255)
75 08          (GLOBAL) REPORT_SIZE        0x08 (8) Number of bits per field
95 07          (GLOBAL) REPORT_COUNT       0x07 (7) Number of fields
B1 02          (MAIN)   FEATURE            0x00000002 (7 fields x 8 bits) 0=Data 1=Variable 0=Absolute 0=NoWrap 0=Linear 0=PrefState 0=NoNull 0=NonVolatile 0=Bitmap
C0           (MAIN)   END_COLLECTION     Application

[INTERFACE 3 HID DESCRIPTOR]
05 01        (GLOBAL) USAGE_PAGE         0x0001 Generic Desktop Page
09 06        (LOCAL)  USAGE              0x00010006 Keyboard (CA=Application Collection)
A1 01        (MAIN)   COLLECTION         0x01 Application (Usage=0x00010006: Page=Generic Desktop Page, Usage=Keyboard, Type=CA)
75 01          (GLOBAL) REPORT_SIZE        0x01 (1) Number of bits per field
95 05          (GLOBAL) REPORT_COUNT       0x05 (5) Number of fields
05 08          (GLOBAL) USAGE_PAGE         0x0008 LED Indicator Page
19 01          (LOCAL)  USAGE_MINIMUM      0x00080001 Num Lock (OOC=On/Off Control)
29 05          (LOCAL)  USAGE_MAXIMUM      0x00080005 Kana (OOC=On/Off Control)
91 02          (MAIN)   OUTPUT             0x00000002 (5 fields x 1 bit) 0=Data 1=Variable 0=Absolute 0=NoWrap 0=Linear 0=PrefState 0=NoNull 0=NonVolatile 0=Bitmap
75 03          (GLOBAL) REPORT_SIZE        0x03 (3) Number of bits per field
95 01          (GLOBAL) REPORT_COUNT       0x01 (1) Number of fields
91 03          (MAIN)   OUTPUT             0x00000003 (1 field x 3 bits) 1=Constant 1=Variable 0=Absolute 0=NoWrap 0=Linear 0=PrefState 0=NoNull 0=NonVolatile 0=Bitmap
75 01          (GLOBAL) REPORT_SIZE        0x01 (1) Number of bits per field
95 08          (GLOBAL) REPORT_COUNT       0x08 (8) Number of fields
15 00          (GLOBAL) LOGICAL_MINIMUM    0x00 (0)
25 01          (GLOBAL) LOGICAL_MAXIMUM    0x01 (1)
05 07          (GLOBAL) USAGE_PAGE         0x0007 Keyboard/Keypad Page
19 E0          (LOCAL)  USAGE_MINIMUM      0x000700E0 Keyboard Left Control (DV=Dynamic Value)
29 E7          (LOCAL)  USAGE_MAXIMUM      0x000700E7 Keyboard Right GUI (DV=Dynamic Value)
81 02          (MAIN)   INPUT              0x00000002 (8 fields x 1 bit) 0=Data 1=Variable 0=Absolute 0=NoWrap 0=Linear 0=PrefState 0=NoNull 0=NonVolatile 0=Bitmap
75 01          (GLOBAL) REPORT_SIZE        0x01 (1) Number of bits per field
95 2E          (GLOBAL) REPORT_COUNT       0x2E (46) Number of fields
15 00          (GLOBAL) LOGICAL_MINIMUM    0x00 (0)
25 01          (GLOBAL) LOGICAL_MAXIMUM    0x01 (1)
05 07          (GLOBAL) USAGE_PAGE         0x0007 Keyboard/Keypad Page
19 04          (LOCAL)  USAGE_MINIMUM      0x00070004 Keyboard a and A (Sel=Selector)
29 31          (LOCAL)  USAGE_MAXIMUM      0x00070031 Keyboard \ and | (Sel=Selector)
81 02          (MAIN)   INPUT              0x00000002 (46 fields x 1 bit) 0=Data 1=Variable 0=Absolute 0=NoWrap 0=Linear 0=PrefState 0=NoNull 0=NonVolatile 0=Bitmap
75 02          (GLOBAL) REPORT_SIZE        0x02 (2) Number of bits per field
95 01          (GLOBAL) REPORT_COUNT       0x01 (1) Number of fields
81 03          (MAIN)   INPUT              0x00000003 (1 field x 2 bits) 1=Constant 1=Variable 0=Absolute 0=NoWrap 0=Linear 0=PrefState 0=NoNull 0=NonVolatile 0=Bitmap
75 01          (GLOBAL) REPORT_SIZE        0x01 (1) Number of bits per field
95 69          (GLOBAL) REPORT_COUNT       0x69 (105) Number of fields
15 00          (GLOBAL) LOGICAL_MINIMUM    0x00 (0)
25 01          (GLOBAL) LOGICAL_MAXIMUM    0x01 (1)
05 07          (GLOBAL) USAGE_PAGE         0x0007 Keyboard/Keypad Page
19 33          (LOCAL)  USAGE_MINIMUM      0x00070033 Keyboard ; and : (Sel=Selector)
29 9B          (LOCAL)  USAGE_MAXIMUM      0x0007009B Keyboard Cancel (Sel=Selector)
81 02          (MAIN)   INPUT              0x00000002 (105 fields x 1 bit) 0=Data 1=Variable 0=Absolute 0=NoWrap 0=Linear 0=PrefState 0=NoNull 0=NonVolatile 0=Bitmap
75 07          (GLOBAL) REPORT_SIZE        0x07 (7) Number of bits per field
95 01          (GLOBAL) REPORT_COUNT       0x01 (1) Number of fields
81 03          (MAIN)   INPUT              0x00000003 (1 field x 7 bits) 1=Constant 1=Variable 0=Absolute 0=NoWrap 0=Linear 0=PrefState 0=NoNull 0=NonVolatile 0=Bitmap
75 01          (GLOBAL) REPORT_SIZE        0x01 (1) Number of bits per field
95 08          (GLOBAL) REPORT_COUNT       0x08 (8) Number of fields
15 00          (GLOBAL) LOGICAL_MINIMUM    0x00 (0)
25 01          (GLOBAL) LOGICAL_MAXIMUM    0x01 (1)
05 07          (GLOBAL) USAGE_PAGE         0x0007 Keyboard/Keypad Page
19 9D          (LOCAL)  USAGE_MINIMUM      0x0007009D Keyboard Prior (Sel=Selector)
29 A4          (LOCAL)  USAGE_MAXIMUM      0x000700A4 Keyboard ExSel (Sel=Selector)
81 02          (MAIN)   INPUT              0x00000002 (8 fields x 1 bit) 0=Data 1=Variable 0=Absolute 0=NoWrap 0=Linear 0=PrefState 0=NoNull 0=NonVolatile 0=Bitmap
75 01          (GLOBAL) REPORT_SIZE        0x01 (1) Number of bits per field
95 2E          (GLOBAL) REPORT_COUNT       0x2E (46) Number of fields
15 00          (GLOBAL) LOGICAL_MINIMUM    0x00 (0)
25 01          (GLOBAL) LOGICAL_MAXIMUM    0x01 (1)
05 07          (GLOBAL) USAGE_PAGE         0x0007 Keyboard/Keypad Page
19 B0          (LOCAL)  USAGE_MINIMUM      0x000700B0 Keypad 00 (Sel=Selector)
29 DD          (LOCAL)  USAGE_MAXIMUM      0x000700DD Keypad Hexadecimal (Sel=Selector)
81 02          (MAIN)   INPUT              0x00000002 (46 fields x 1 bit) 0=Data 1=Variable 0=Absolute 0=NoWrap 0=Linear 0=PrefState 0=NoNull 0=NonVolatile 0=Bitmap
75 02          (GLOBAL) REPORT_SIZE        0x02 (2) Number of bits per field
95 01          (GLOBAL) REPORT_COUNT       0x01 (1) Number of fields
81 03          (MAIN)   INPUT              0x00000003 (1 field x 2 bits) 1=Constant 1=Variable 0=Absolute 0=NoWrap 0=Linear 0=PrefState 0=NoNull 0=NonVolatile 0=Bitmap
C0           (MAIN)   END_COLLECTION     Application

[INTERFACE 4 HID DESCRIPTOR]
05 0C        (GLOBAL) USAGE_PAGE         0x000C Consumer Device Page
09 01        (LOCAL)  USAGE              0x000C0001 Consumer Control (CA=Application Collection)
A1 01        (MAIN)   COLLECTION         0x01 Application (Usage=0x000C0001: Page=Consumer Device Page, Usage=Consumer Control, Type=CA)
75 08          (GLOBAL) REPORT_SIZE        0x08 (8) Number of bits per field
95 01          (GLOBAL) REPORT_COUNT       0x01 (1) Number of fields
15 01          (GLOBAL) LOGICAL_MINIMUM    0x01 (1)
26 EA00        (GLOBAL) LOGICAL_MAXIMUM    0x00EA (234)
05 0C          (GLOBAL) USAGE_PAGE         0x000C Consumer Device Page
19 01          (LOCAL)  USAGE_MINIMUM      0x000C0001 Consumer Control (CA=Application Collection)
2A EA00        (LOCAL)  USAGE_MAXIMUM      0x000C00EA Volume Decrement (RTC=Re-trigger Control)
81 00          (MAIN)   INPUT              0x00000000 (1 field x 8 bits) 0=Data 0=Array 0=Absolute 0=Ignored 0=Ignored 0=PrefState 0=NoNull
C0           (MAIN)   END_COLLECTION     Application

[INTERFACE 5 HID DESCRIPTOR]
05 01        (GLOBAL) USAGE_PAGE         0x0001 Generic Desktop Page
09 05        (LOCAL)  USAGE              0x00010005 Game Pad (CA=Application Collection)
A1 01        (MAIN)   COLLECTION         0x01 Application (Usage=0x00010005: Page=Generic Desktop Page, Usage=Game Pad, Type=CA)
A1 00          (MAIN)   COLLECTION         0x00 Physical (Usage=0x0: Page=, Usage=, Type=)
05 09            (GLOBAL) USAGE_PAGE         0x0009 Button Page
19 01            (LOCAL)  USAGE_MINIMUM      0x00090001 Button 1 Primary/trigger (MULTI=Selector, On/Off, Momentary, or One Shot)
29 0C            (LOCAL)  USAGE_MAXIMUM      0x0009000C Button 12 (MULTI=Selector, On/Off, Momentary, or One Shot)
15 00            (GLOBAL) LOGICAL_MINIMUM    0x00 (0)
25 01            (GLOBAL) LOGICAL_MAXIMUM    0x01 (1)
95 0C            (GLOBAL) REPORT_COUNT       0x0C (12) Number of fields
75 01            (GLOBAL) REPORT_SIZE        0x01 (1) Number of bits per field
81 02            (MAIN)   INPUT              0x00000002 (12 fields x 1 bit) 0=Data 1=Variable 0=Absolute 0=NoWrap 0=Linear 0=PrefState 0=NoNull 0=NonVolatile 0=Bitmap
75 04            (GLOBAL) REPORT_SIZE        0x04 (4) Number of bits per field
95 01            (GLOBAL) REPORT_COUNT       0x01 (1) Number of fields
81 03            (MAIN)   INPUT              0x00000003 (1 field x 4 bits) 1=Constant 1=Variable 0=Absolute 0=NoWrap 0=Linear 0=PrefState 0=NoNull 0=NonVolatile 0=Bitmap
05 01            (GLOBAL) USAGE_PAGE         0x0001 Generic Desktop Page
09 30            (LOCAL)  USAGE              0x00010030 X (DV=Dynamic Value)
09 31            (LOCAL)  USAGE              0x00010031 Y (DV=Dynamic Value)
09 32            (LOCAL)  USAGE              0x00010032 Z (DV=Dynamic Value)
09 33            (LOCAL)  USAGE              0x00010033 Rx (DV=Dynamic Value)
09 34            (LOCAL)  USAGE              0x00010034 Ry (DV=Dynamic Value)
09 35            (LOCAL)  USAGE              0x00010035 Rz (DV=Dynamic Value)
16 0080          (GLOBAL) LOGICAL_MINIMUM    0x8000 (-32768)
26 FF7F          (GLOBAL) LOGICAL_MAXIMUM    0x7FFF (32767)
75 10            (GLOBAL) REPORT_SIZE        0x10 (16) Number of bits per field
95 06            (GLOBAL) REPORT_COUNT       0x06 (6) Number of fields
81 02            (MAIN)   INPUT              0x00000002 (6 fields x 16 bits) 0=Data 1=Variable 0=Absolute 0=NoWrap 0=Linear 0=PrefState 0=NoNull 0=NonVolatile 0=Bitmap
05 02            (GLOBAL) USAGE_PAGE         0x0002 Simulation Controls Page
09 BB            (LOCAL)  USAGE              0x000200BB Throttle (DV=Dynamic Value)
09 B8            (LOCAL)  USAGE              0x000200B8 Elevator (DV=Dynamic Value)
09 C4            (LOCAL)  USAGE              0x000200C4 Accelerator (DV=Dynamic Value)
09 BA            (LOCAL)  USAGE              0x000200BA Rudder (DV=Dynamic Value)
09 C0            (LOCAL)  USAGE              0x000200C0 Trigger (MC=Momentary Control)
15 00            (GLOBAL) LOGICAL_MINIMUM    0x00 (0)
26 FF00          (GLOBAL) LOGICAL_MAXIMUM    0x00FF (255)
75 08            (GLOBAL) REPORT_SIZE        0x08 (8) Number of bits per field
95 05            (GLOBAL) REPORT_COUNT       0x05 (5) Number of fields
81 02            (MAIN)   INPUT              0x00000002 (5 fields x 8 bits) 0=Data 1=Variable 0=Absolute 0=NoWrap 0=Linear 0=PrefState 0=NoNull 0=NonVolatile 0=Bitmap
C0             (MAIN)   END_COLLECTION     Physical
C0           (MAIN)   END_COLLECTION     Application

[INTERFACE 6 HID DESCRIPTOR]
06 FFFF      (GLOBAL) USAGE_PAGE         0xFFFF Vendor-defined
09 01        (LOCAL)  USAGE              0xFFFF0001 
A1 01        (MAIN)   COLLECTION         0x01 Application (Usage=0xFFFF0001: Page=Vendor-defined, Usage=, Type=)
09 02          (LOCAL)  USAGE              0xFFFF0002 
09 03          (LOCAL)  USAGE              0xFFFF0003 
15 00          (GLOBAL) LOGICAL_MINIMUM    0x00 (0)
26 FF00        (GLOBAL) LOGICAL_MAXIMUM    0x00FF (255)
75 08          (GLOBAL) REPORT_SIZE        0x08 (8) Number of bits per field
95 20          (GLOBAL) REPORT_COUNT       0x20 (32) Number of fields
81 02          (MAIN)   INPUT              0x00000002 (32 fields x 8 bits) 0=Data 1=Variable 0=Absolute 0=NoWrap 0=Linear 0=PrefState 0=NoNull 0=NonVolatile 0=Bitmap
C0           (MAIN)   END_COLLECTION     Application

```

**Comment by J2Kbr** [comment]
Fixed on firmware 0.98B-72

**Comment by paname75** [comment]
for this specific keyboard , 
the keyboard when plugged to titan2 even  with last update 98.1.72 doesnt output some keys,  like : 

- spacebar
- - = { }

here is my descriptor fetch 


[USB DESCRIPTOR]
12 01 00 02 00 00 00 40 EB 03 01 FF 11 01 01 02 00 01 09 02 CE 00 07 01 00 A0 C8 09 04 00 00 02 FF 5D 01 00 11 21 00 01 01 25 81 14 00 00 00 00 13 02 08 00 00 07 05 81 03 20 00 04 07 05 02 03 20 00 08 09 04 01 00 01 03 01 01 00 09 21 11 01 00 01 22 3B 00 07 05 83 03 08 00 0A 09 04 02 00 02 03 00 00 00 09 21 11 01 00 01 22 35 00 07 05 85 03 40 00 04 07 05 04 03 40 00 04 09 04 03 00 01 03 00 01 00 09 21 11 01 00 01 22 7B 00 07 05 86 03 40 00 01 09 04 04 00 01 03 00 01 00 09 21 11 01 00 01 22 19 00 07 05 87 03 08 00 10 09 04 05 00 01 03 00 00 00 09 21 11 01 00 01 22 51 00 07 05 88 03 20 00 08 09 04 06 00 01 03 00 00 00 09 21 11 01 00 01 22 17 00 07 05 89 03 40 00 04 
 
[STRING 1 DESCRIPTOR]
00 00 00 02 00 00 02 01 80 06 03 01 04 09 00 10 
10 03 57 00 6F 00 6F 00 74 00 69 00 6E 00 67 00 
 
[STRING 2 DESCRIPTOR]
00 00 00 02 00 00 02 01 80 06 03 02 04 09 00 16 
16 03 57 00 6F 00 6F 00 74 00 69 00 6E 00 67 00 4F 00 6E 00 65 00 
 
[INTERFACE 1 HID DESCRIPTOR]
00 00 00 02 00 00 02 01 81 06 22 00 00 01 00 3B 
05 01 09 06 A1 01 05 07 19 E0 29 E7 15 00 25 01 75 01 95 00 81 02 81 01 19 00 29 65 15 00 25 65 75 08 95 00 81 00 05 08 19 01 29 05 15 00 25 01 75 01 95 05 91 02 95 03 91 01 C0 
 
[INTERFACE 2 HID DESCRIPTOR]
00 00 00 02 00 00 02 01 81 06 22 00 00 02 00 35 
06 37 13 09 01 A1 01 09 02 09 03 15 00 26 FF 00 75 08 95 80 81 02 09 04 09 05 15 00 26 FF 00 75 08 95 80 91 02 09 06 09 07 15 00 26 FF 00 75 08 95 07 B1 02 C0 
 
[INTERFACE 3 HID DESCRIPTOR]
00 00 00 02 00 00 02 01 81 06 22 00 00 03 00 7B 
05 01 09 06 A1 01 75 01 95 05 05 08 19 01 29 05 91 02 75 03 95 01 91 03 75 01 95 08 15 00 25 01 05 07 19 E0 29 E7 81 02 75 01 95 2E 15 00 25 01 05 07 19 04 29 31 81 02 75 02 95 01 81 03 75 01 95 69 15 00 25 01 05 07 19 33 29 9B 81 02 75 07 95 01 81 03 75 01 95 08 15 00 25 01 05 07 19 9D 29 A4 81 02 75 01 95 2E 15 00 25 01 05 07 19 B0 29 DD 81 02 75 02 95 01 81 03 C0 
 
[INTERFACE 4 HID DESCRIPTOR]
00 00 00 02 00 00 02 01 81 06 22 00 00 04 00 19 
05 0C 09 01 A1 01 75 08 95 01 15 01 26 EA 00 05 0C 19 01 2A EA 00 81 00 C0 
 
[INTERFACE 5 HID DESCRIPTOR]
00 00 00 02 00 00 02 01 81 06 22 00 00 05 00 51 
05 01 09 05 A1 01 A1 00 05 09 19 01 29 0C 15 00 25 01 95 0C 75 01 81 02 75 04 95 01 81 03 05 01 09 30 09 31 09 32 09 33 09 34 09 35 16 00 80 26 FF 7F 75 10 95 06 81 02 05 02 09 BB 09 B8 09 C4 09 BA 09 C0 15 00 26 FF 00 75 08 95 05 81 02 C0 C0 
 
[INTERFACE 6 HID DESCRIPTOR]
00 00 00 02 00 00 02 01 81 06 22 00 00 06 00 17 
06 FF FF 09 01 A1 01 09 02 09 03 15 00 26 FF 00 75 08 95 20 81 02 C0
